server/PokerTableObjects/PokerHandRankCalculator.js

```diff
@@ -0,0 +1,425 @@
+            let split = communityCards[i].split("_");
+            let number;
+            if(split[0] === 'A'){
+                number = 1;
+            }
+            else if(split[0] === 'J'){
+                number = 11;
+            }
+            else if(split[0] === 'Q'){
+                number = 12;
+            }
+            else if(split[0] === 'K'){
+                number = 13;
+            }
+            else{
+                number = parseInt(split[0]);
+            }
+            let suit = split[1];
+            cards[i] = new Card(number, suit);
+        }
+        for(let i = 0; i < 2; i++){ // convert the two player cards into objects the calcultor can use
+            let split = playerCards[i].split("_");
+            let number;
+            if(split[0] === 'A'){
+                number = 1;
+            }
+            else if(split[0] === 'J'){
+                number = 11;
+            }
+            else if(split[0] === 'Q'){
+                number = 12;
+            }
+            else if(split[0] === 'K'){
+                number = 13;
+            }
+            else{
+                number = parseInt(split[0]);
+            }
+            let suit = split[1];
+            cards[i+5] = new Card(number, suit);
+        }
+        console.log(JSON.stringify(cards));
+        // we will iterate through the functions until one returns a positive value
+        /*
+        let allFunctions = [
+            this.handStraightFlushValue,
+            this.handQuadsValue,
+            this.handFullHouseValue,
+            this.handFlushValue,
+            this.handStraightValue,
+            this.handThreeOfAKindValue,
+            this.handTwoPairValue,
+            this.handSinglePairValue,
+            this.handHighCardValue
+        ];
+        */
+        // number refers to how many cards to include in the highcard calculation
+        // condition refers to how that function is to return that result, each function
+        // uses condition to its own definition
+        let setUps = [{},{},{},{condition: false}, {condition: true}, {number: 2}, {number: 1}, {number: 3}, {number: 5, condition: true}];
+        
+        let result = -1;
+        /*
+        for(let i = 0; i < allFunctions.length; i++){
+            result = allFunctions[i](cards, setUps[i]); // for some reason this is not working, claims functions are not defined
+            if(result != -1){
+                return result;
+            }
+        }
+        */
+
+
+
+        // NOTE: up until this point I have not found a way to combine all these functions into an array that i can iterate
+        // through, the server crashes (claiming it cannot find my functions and thus I reverted the code back to the messy)
+        // lines of 'if statments' below, do not remove the commented out code above as I might come back to this problem at a
+        // later date and try to fix it.
```

Comment:
Is there a point in checking broken code into master?  As long as this branch isn't deleted, we should be able to pull the code in the future if we feel we need to revisit this issue.  If we're planning to come back to this, it should be tracked in trello so we don't forget about it.